frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardView.stories.tsx

```diff
@@ -0,0 +1,227 @@
+import { useEffect, type ComponentProps } from "react";
+import { Provider } from "react-redux";
+
+import { getStore } from "__support__/entities-store";
+import { getNextId } from "__support__/utils";
+import { NumberColumn, StringColumn } from "__support__/visualizations";
+import { explicitSizeRefreshModeContext } from "metabase/components/ExplicitSize/context";
+import { publicReducers } from "metabase/reducers-public";
+import TABLE_RAW_SERIES from "metabase/visualizations/components/TableSimple/stories-data/table-simple-orders-with-people.json";
+import {
+  createMockCard,
+  createMockColumn,
+  createMockDashboard,
+  createMockDashboardCard,
+  createMockDataset,
+  createMockDatasetData,
+  createMockParameter,
+} from "metabase-types/api/mocks";
+import {
+  createMockDashboardState,
+  createMockState,
+} from "metabase-types/store/mocks";
+
+import { PublicOrEmbeddedDashboardView } from "./PublicOrEmbeddedDashboardView";
+
+export default {
+  title: "embed/PublicOrEmbeddedDashboardView",
+  component: PublicOrEmbeddedDashboardView,
+  decorators: [
+    ReduxDecorator,
+    FasterExplicitSizeUpdateDecorator,
+    WaitForResizeToStopDecorator,
+  ],
+};
+
+function ReduxDecorator(Story: Story) {
+  return (
+    <Provider store={store}>
+      <Story />
+    </Provider>
+  );
+}
+
+function FasterExplicitSizeUpdateDecorator(Story: Story) {
+  return (
+    <explicitSizeRefreshModeContext.Provider value="none">
+      <Story />
+    </explicitSizeRefreshModeContext.Provider>
+  );
+}
+
+/**
+ * This is an arbitrary number, it should be big enough to pass CI tests.
+ * This value works together with FasterExplicitSizeUpdateDecorator which
+ * make sure we finish resizing any ExplicitSize components the fastest.
+ */
+const TIME_UNTIL_ALL_ELEMENTS_STOP_RESIZING = 1000;
+function WaitForResizeToStopDecorator(Story: Story) {
+  const asyncCallback = createAsyncCallback();
+  useEffect(() => {
+    setTimeout(asyncCallback, TIME_UNTIL_ALL_ELEMENTS_STOP_RESIZING);
+  }, [asyncCallback]);
+
+  return <Story />;
+}
+
+const DASHBOARD_ID = getNextId();
+const DASHCARD_BAR_ID = getNextId();
+const DASHCARD_TABLE_ID = getNextId();
+const CARD_BAR_ID = getNextId();
+const CARD_TABLE_ID = getNextId();
+const TAB_ID = getNextId();
+const PARAMETER_ID = "param-hex";
+const initialState = createMockState({
+  dashboard: createMockDashboardState({
+    dashcardData: {
+      [DASHCARD_BAR_ID]: {
+        [CARD_BAR_ID]: createMockDataset({
+          data: createMockDatasetData({
+            cols: [
+              createMockColumn(StringColumn({ name: "Dimension" })),
+              createMockColumn(NumberColumn({ name: "Count" })),
+            ],
+            rows: [
+              ["foo", 1],
+              ["bar", 2],
+            ],
+          }),
+        }),
+      },
+      [DASHCARD_TABLE_ID]: {
+        // Couldn't really figure out the type here.
+        [CARD_TABLE_ID]: createMockDataset(TABLE_RAW_SERIES[0] as any),
```

Comment:
`TABLE_RAW_SERIES[0]` should be `SingleSeries` from `frontend/src/metabase-types/api/dataset.ts`

Reply:
I couldn't really figure out how to make this work.
![image](https://github.com/metabase/metabase/assets/1937582/4784c00d-2adf-4043-b27a-6e66fb6e0ad0)

I've checked the data we store in Redux, it seems we didn't make any atypical transformation to the series.

Would you mind give me a guidance on this?

Reply:
Ah, it will be tricky to make these json imports match types I guess, never mind then